amanuensis/resources/filter_sets/__init__.py

```diff
@@ -0,0 +1,167 @@
+            logger.info(f"The filter {filter_set_name} contains the element {key} which is not a valid element in the data dictionary")
+            return False
+        
+        try: 
+            selected_type = selected_value["__type"]
+            if selected_type == "OPTION":
+                selected_values = selected_value["selectedValues"]
+        except Exception as e:
+            logger.info(f"The filter {filter_set_name} is not in a valid format")
+            return False
+
+        if selected_type == "OPTION":
+            if "enum" in es_to_dd_map[key]:
+                for value in selected_values:
+                    if value not in es_to_dd_map[key]["enum"]:
+                        logger.info(f"The filter {filter_set_name} has a selected value {value} for the filter {key} which is not a valid element in the data dictionary")
+                        return False
+                    
+        elif selected_type != "RANGE":
+            logger.info(f"The filter {filter_set_name} has a selected filter {key} which is not a valid format {selected_value}")
+            return False
+        
+    logger.info(f"The filter {filter_set_name} is valid for the data dictionary")   
+    return True   
+
+
+def check_filter_sets(session, es_to_dd_map_file_name="es_to_dd_map.json", portal_config_file_name="gitops.json"):
+
+    es_to_dd_map = _load_data_files(es_to_dd_map_file_name)
+    
+    portal_config = _load_data_files(portal_config_file_name)
+
+    explorer_selectable_values = _get_explorer_selectable_values(portal_config)
+
+    filter_sets = get_filter_sets(session, filter_by_active=False, filter_by_source_type=False)
+
+    for filter_set in filter_sets:
+
+        if "value" in filter_set.filter_object:
+            
+            is_valid = _check_portal_config(filter_set, explorer_selectable_values) 
+
+            if is_valid:
+
+                is_valid = _check_es_to_dd_map(filter_set, es_to_dd_map)
+        else:
+            is_valid = False
+
+
+        if not is_valid:
+
+            if filter_set.is_valid == True:
+
+                logger.info(f"name: {filter_set.name} id: {filter_set.id} is invalid but marked as valid moving to invalid")
+
+                update_filter_set(session, filter_set=filter_set, is_valid=False)
+
+            else:
+
+                logger.info(f"name: {filter_set.name} id: {filter_set.id} is invalid and already marked as invalid doing nothing")
+
+        if is_valid:
+
+            if filter_set.is_valid == False:
+
+                logger.info(f"name: {filter_set.name} id: {filter_set.id} is valid but marked as invalid moving back to valid")
+
+                update_filter_set(session, filter_set=filter_set, is_valid=True)
+
+            else:
+
+                logger.info(f"name: {filter_set.name} id: {filter_set.id} is valid and already marked as valid doing nothing")
```

Comment:
The update filterset method should reset this to True. The scenario is when a filterset is set to invalid by this. Then the user sees that is invalid and updates the filter-set in the exploration page and saves it. The save there should reset the is_valid to True.

Reply:
I modified the update filter-set method so that when the method is called if a graphql or filter-set object is passed it automatically sets it to True for the condition you mention but does nothing in case an update is just changing the name of the filter-set